juju/juju-restore
# Manage Juju agents.

## Description of change

As part of the Level 0 pre-check, we need to bring down Juju agents on primary controller node as well as on the secondary controller nodes if they are managed by 'juju-restore' and not the user.

Since we will eventually need to start these stopped agents up, the PR also allows to restart the agents if --rs option is provided. This option will be removed once the actual restore will be implemented since it will be a normal flow of operation - stop agents, restore, start agents again.
 
To correctly name juju agent (juud-machine-*), we need to know Juju Machine ID. This PR also adds this information to replica set member.


## QA steps

1. bootstrap
2. enable-ha
3. run 'juju-restore' on a primary
4. Once the prompts are answered, all juju agents should be stopped. I checked with 'systemctl status jujud-machine-*' on each controller machine.
5. If desired, re-run 'juju-restore' with --rs to restart Juju agents.
